src/main/java/org/folio/consortia/repository/UserTenantRepository.java

```diff
@@ -33,4 +34,11 @@ public interface UserTenantRepository extends JpaRepository<UserTenantEntity, UU
   @Modifying
   @Query("DELETE FROM UserTenantEntity ut WHERE ut.userId= ?1 AND ut.isPrimary= true")
   void deleteByUserIdAndIsPrimaryTrue(UUID userId);
+
+  @Query("SELECT ut FROM UserTenantEntity ut WHERE ut.userId NOT IN (SELECT ut.userId FROM UserTenantEntity ut WHERE ut.isPrimary=true)")
+  Optional<List<UserTenantEntity>> getByUserIdAndIsPrimaryFalse();
+
+  @Modifying
+  @Query("DELETE FROM UserTenantEntity ut WHERE ut.userId NOT IN (SELECT ut.userId FROM UserTenantEntity ut WHERE ut.isPrimary=true)")
+  void deleteByUserIdAndIsPrimaryFalse();
```

Comment:
You need to include param userId, because now we are deleting not in scheduled job, but when precessing USER_DELETED event from kafka, so we should delete shadow users only for that user id

Reply:
added userId param